lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -440,6 +440,13 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
   DeclName Name = UDRE->getName();
   SourceLoc Loc = UDRE->getLoc();
 
+  // 'Any' is the empty protocol composition; resolve it before doing an
+  // unqualified lookup
+  if (Name.getBaseName() == Context.Id_Any) {
```

Comment:
This isn't correct; `X.Any` should still be a valid name for a nested type if it was declared as follows:

```swift
struct Outer {
  typealias `Any` = Int
}
```

I think it's okay, though, because you already eagerly resolved this in the parser.